File: anacondacode/tensorflow_environment/old_code/计算方法/homework3_4.py
# -*- coding: utf-8 -*-
"""
Created on Sat Sep 21 12:11:54 2019

@author: LJZ
"""

#二分法
import numpy as np
import matplotlib.pyplot as plt
import math
e=math.e
f=lambda x:2*x+1-e**x
x=np.linspace(-0.8,2.5,101)
plt.plot(x,f(x))
#hold on
plt.plot(x,np.zeros(x.shape))
plt.draw
a=x[0]
b=x[-1]
c=(a+b)/2
while(abs(f(c))>1e-4):
    if(f(c)>0):
        a=c
    else:
        b=c
    plt.plot(c,f(c),'*')
    plt.draw
    #pause(1)
    
    c=(a+b)/2

File: matlab_code/oldcode/homework3_4.m
%¶þ·Ö·¨
f=@(x)2*x+1-exp(x);
x=linspace(-0.8,2.5,101);
plot(x,f(x));
hold on;plot(x,zeros(1,101));
a=x(1);b=x(end);
c=(a+b)/2;
while(abs(f(c))>1e-4)
    if(f(c)>0)
        a=c;
    else
        b=c;
    end
    hold on;scatter(c,f(c),'*','LineWidth',5);pause(1);   
    c=(a+b)/2;
end